certification/engine/engine.go

```diff
@@ -28,137 +28,107 @@ type CheckEngine interface {
 	Results(context.Context) runtime.Results
 }
 
-func NewForConfig(config runtime.Config) (CheckEngine, error) {
-	if len(config.EnabledChecks) == 0 {
-		// refuse to run if the user has not specified any checks
-		return nil, fmt.Errorf("no checks have been enabled")
-	}
-
-	checks := make([]certification.Check, 0, len(config.EnabledChecks))
-	for _, checkString := range config.EnabledChecks {
-		check := queryChecks(checkString)
-		if check == nil {
-			err := fmt.Errorf("requested check not found: %s", checkString)
-			return nil, err
-		}
-
-		checks = append(checks, check)
+func NewForConfig(cfg certification.Config) (CheckEngine, error) {
+	checks, err := initializeChecks(context.Background(), cfg.Policy(), cfg)
```

Comment:
Context should be passed in here. We are now creating the context in cmd/root.go. Anything that needs a context now should be getting it from there.

Reply:
Done. I passed `context.TODO()` in tests, and `cmd.Context()` in other cases.